frame/vamm/src/tests/swap.rs

```diff
@@ -463,12 +488,23 @@ proptest! {
 			vamm_state,
 			swap_config,
 			|swap_config| {
-				// For event emission.
-				run_for_seconds(vamm_state.twap_period);
-				if TestPallet::swap(&swap_config).is_ok() {
-					// Ensure twap was updated
-					let vamm_state_after = TestPallet::get_vamm(0).unwrap();
-					assert_ne!(vamm_state.twap_timestamp, vamm_state_after.twap_timestamp);
+				run_for_seconds(delta);
+				let swap_output = TestPallet::swap(&swap_config);
+				match vamm_state.twap_period.cmp(&delta) {
+					Less => {
+						if swap_output.is_ok() {
+							// Ensure twap was updated
+							let vamm_state_after = TestPallet::get_vamm(0).unwrap();
+							assert_ne!(vamm_state.twap_timestamp, vamm_state_after.twap_timestamp);
+						}
+					},
+					_ => {
+						if swap_output.is_ok() {
+							// Ensure twap was *not* updated
+							let vamm_state_after = TestPallet::get_vamm(0).unwrap();
+							assert_eq!(vamm_state.twap_timestamp, vamm_state_after.twap_timestamp);
```

Comment:
Isn't `twap_period` only used to compute the weights in the weighted average? Why is it that if more than `twap_period` has passed, the TWAP isn't updated? The more time passes, the more aggressive we should be with the update (less weight to older TWAP value, more to current price value)

Reply:
You're correct.
Fixed it in recent commits 👍